glfw/src/Window.zig

```diff
@@ -1361,9 +1416,15 @@ pub inline fn setPosCallback(self: Window, callback: ?fn (window: Window, xpos:
     var internal = self.getInternal();
     internal.setPosCallback = callback;
     _ = c.glfwSetWindowPosCallback(self.handle, if (callback != null) setPosCallbackWrapper else null);
-    getError() catch unreachable; // Only error 'GLFW_NOT_INITIALIZED' is impossible
+    getError() catch |err| return switch (err) {
+        Error.NotInitialized => unreachable,
+        else => unreachable,
+    };
 }
 
+// TODO: don't the calls to `from` in these `set*CallbackWrapper` functions cause a leak?
+// is this what's being reported by valgrind in [#60](https://github.com/hexops/mach/issues/60)?
```

Comment:
Hmm, `from` is only going to allocate a user pointer for the window if one doesn't already exist, though.

That is, as long as the callback wrapper is invoked with the `handle` of a window created by mach/glfw (and not some other GLFW window created outside mach/glfw), then inside of `from` `glfwGetWindowUserPointer` should return a non-null pointer and no leak should happen.

also note that inside of `Window.destroy` we only destroy the user pointer object _after_ we've destroyed the window, so no calls to this callback should be issued afterwards either. https://github.com/hexops/mach/blob/233fc53a17d16ab4901bda9f7bd332663e750c75/glfw/src/Window.zig#L448-L450

So I don't think any leak is possible here, but if you think otherwise I am super keen to hear about it!

Reply:
Maybe we can/should use GeneralPurposeAllocator here to verify we're not leaking

Reply:
You're right, there is no leak, not directly observable from what I've tried at least; false alarm on my part.